### PR TITLE
collateral support in the builder

### PR DESCRIPTION
**Pull request by vsubhuman**

### CODE REVIEW

**Review by vsubhuman — COMMENTED** [review]
/check

**Review by vsubhuman — COMMENTED** [review]
/check

**Review by vsubhuman — COMMENTED** [review]
/check

File: rust/src/tx_builder/tx_inputs_builder.rs [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+                vkeys: BTreeSet::new(),
+                scripts: LinkedHashMap::new(),
+                bootstraps: BTreeSet::new(),
+            },
+        }
+    }
+
+    fn push_input(&mut self, e: (TxBuilderInput, Option<ScriptHash>)) {
+        self.inputs.push(e);
+        self.inputs.sort_by(|(a, _), (b, _)| a.input.cmp(&b.input));
```
Comment (lisicky) [inline rust/src/tx_builder/tx_inputs_builder.rs:135]: If we have sort for each push, can we use smth like a BTreeMap or  binary heap ? 
https://doc.rust-lang.org/std/collections/struct.BTreeMap.html
https://doc.rust-lang.org/stable/std/collections/struct.BinaryHeap.html

Comment (vsubhuman) [reply rust/src/tx_builder/tx_inputs_builder.rs:135]: good point, will check

Comment (vsubhuman) [reply rust/src/tx_builder/tx_inputs_builder.rs]: done

**Review by vsubhuman — COMMENTED** [review]
/check

**Review by lisicky — APPROVED** [review]

**Review by vsubhuman — COMMENTED** [review]
/check

**Review by vsubhuman — COMMENTED** [review]
/check

**Pull request merged**